### DISCUSSION TITLE
Question about using the various CNI

### DISCUSSION THREAD

**Discussion by lchunleo** [rancher/rke2#5362] [Q&A] [answered]
Hi 

 Am starting out to learn and use rke2. However, i like to have some advice to use rke2 in air gap. I understand that i can install different CNI from the asset download. However, i wish to know if e.g. i install canal as my cni and in future i wish to add calico or weavenet , do i need to uninstall everything and do a reinstallation?. Or i have to use the multux  to install before i can add various CNI into it. Thanks

**Comment by manuelbuil** [comment, marked as answer]
Changing CNIs after the cluster has been deployed is not something supported in Kubernetes. So, yes, you'll need to reinstall

**Comment by lchunleo** [comment]
> Changing CNIs after the cluster has been deployed is not something supported in Kubernetes. So, yes, you'll need to reinstall 

Reading the air gap instructions, can i check if my understanding is correct to deploy rke2 with weavenet.. https://docs.rke2.io/install/airgap

1) Go to the asset and download the 1) sha256-amd..txt 2) rke2-images-linux-amd64 3) rke2-images-core-linux-amd64 4)install.sh
2) Based on the required own CNI and follow the instructions there? (or need to do something to link up the rke2 with the new CNI?)

Thanks
